CondTools/SiStrip/plugins/SiStripApvGainFromFileBuilder.cc

```diff
@@ -0,0 +1,637 @@
+    // uncabled. This is worth only if we decide not tu put the record
+    // in the DB for the uncabled det_id.
+    //if( !detCabling_->IsConnected(it.first) ) {
+    //
+    //  continue;
+    //}
+
+    //Gather the APV online id
+    std::vector<std::pair<int, float>> tickmark_for_detId(it.second.nApvs, std::pair<int, float>(-1, 999999.));
+    for (unsigned int ca = 0; ca < connection.size(); ca++) {
+      if (connection[ca] != nullptr) {
+        uint16_t id1 = (connection[ca])->i2cAddr(0) % 32;
+        uint16_t id2 = (connection[ca])->i2cAddr(1) % 32;
+        tickmark_for_detId[online2offline(id1, it.second.nApvs)].first = id1;
+        tickmark_for_detId[online2offline(id2, it.second.nApvs)].first = id2;
+      }
+    }
+    gain_from_maps(it.first, it.second.nApvs, tickmark_for_detId);
+    std::vector<float> theSiStripVector;
+
+    // Fill the gain in the DB object, apply the logic for the dummy values
+    for (unsigned short j = 0; j < it.second.nApvs; j++) {
+      Summary summary;
+      summary.det_id = it.first;
+      summary.offlineAPV_id = j;
+      summary.onlineAPV_id = tickmark_for_detId.at(j).first;
+      summary.is_connected = false;
+      summary.FED_id = -1;
+      summary.FED_ch = -1;
+      summary.i2cAdd = -1;
+
+      for (unsigned int ca = 0; ca < connection.size(); ca++) {
+        if (connection[ca] != nullptr && (connection[ca])->i2cAddr(j % 2) % 32 == summary.onlineAPV_id) {
+          summary.is_connected = (connection[ca])->isConnected();
+          summary.FED_id = (connection[ca])->fedId();
+          summary.FED_ch = (connection[ca])->fedCh();
+          summary.i2cAdd = (connection[ca])->i2cAddr(j % 2);
+        }
+      }
+
+      try {
+        float gain = tickmark_for_detId[j].second;
+        summary.gain_from_scan = gain;
+        LogTrace("SiStripApvGainFromFileBuilder")
+            << it.first << "  " << std::setw(3) << j << "   " << std::setw(3) << connection.size() << "   "
+            << std::setw(3) << summary.onlineAPV_id << "    " << std::setw(3) << summary.FED_id << "   " << std::setw(3)
+            << summary.FED_ch << "   " << std::setw(3) << summary.i2cAdd << "   " << std::setw(7)
+            << summary.gain_from_scan << std::endl;
+
+        if (gain != 999999.) {
+          summary.is_scanned = true;
+          if (gain > gainThreshold_) {
+            if (doGainNormalization_) {
+              // divide the tickmark by the normalization factor (640)
+              gain /= k_GainNormalizationFactor;
+            }
+            summary.gain_in_db = gain;
+            if (!summary.is_connected)
+              ex_summary_.push_back(summary);
+            else
+              summary_.push_back(summary);
+          } else {
+            if (gain == 0.) {
+              if (putDummyIntoOffChannels_)
+                summary.gain_in_db = dummyAPVGain_;
+              else
+                summary.gain_in_db = 0.;
```

Comment:
```suggestion
                summary.gain_in_db = (putDummyIntoOffChannels_? dummyAPVGain_ : 0.);
```
is more compact.
This is not a request, just a suggestion: apply only if you like it and if there are other updates to be also applied to the code. (Same for the other similar cases)

Reply:
addressed at https://github.com/cms-sw/cmssw/pull/37066/commits/46ef1337867f703f62e46a989fe308fa54e61dcc